compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.td

```diff
@@ -202,5 +312,152 @@ def IREEGPU_MmaScheduleAttr : AttrDef<IREEGPU_Dialect, "MMASchedule"> {
+    workgroup is scheduled onto; that is, a compute unit for AMD GPUs or a
+    streaming multiprocessor for NVIDIA GPUs.
+  }];
+
+  let mnemonic = "target_chip";
+  let cppNamespace = "::mlir::iree_compiler::IREE::GPU";
+
+  let parameters = (ins
+    "uint32_t":$core_count
+  );
+
+  let assemblyFormat = "`<` struct(params) `>`";
+}
+
+//===----------------------------------------------------------------------===//
+// GPU Target Attributes
+//===----------------------------------------------------------------------===//
+
+def IREEGPU_TAPI_CUDA    : I32EnumAttrCase<"CUDA",    0, "cuda">;
+def IREEGPU_TAPI_DirectX : I32EnumAttrCase<"DirectX", 1, "directx">;
+def IREEGPU_TAPI_HIP     : I32EnumAttrCase<"HIP",     2, "hip">;
+def IREEGPU_TAPI_Metal   : I32EnumAttrCase<"Metal",   3, "metal">;
+def IREEGPU_TAPI_Vulkan  : I32EnumAttrCase<"Vulkan",  4, "vulkan">;
+def IREEGPU_TAPI_WebGPU  : I32EnumAttrCase<"WebGPU",  5, "webgpu">;
+
+def IREEGPU_TargetAPI : I32EnumAttr<
+  "TargetAPI", "Supported target APIs",
+  [IREEGPU_TAPI_CUDA, IREEGPU_TAPI_DirectX, IREEGPU_TAPI_HIP,
+   IREEGPU_TAPI_Metal, IREEGPU_TAPI_Vulkan, IREEGPU_TAPI_WebGPU]> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::GPU";
+  let genSpecializedAttr = 0;
+}
+
+def IREEGPU_TargetAttr : AttrDef<IREEGPU_Dialect, "Target"> {
+  let summary = "Full GPU target attribute";
+  let description = [{
+    This attributes describes a full GPU target. It contains a few fields:
+    * The target GPU API, e.g., cuda, hip, vulkan
+    * The canonical target architecture for compilation, e.g., sm_80 for
+      cuda, gfx942 for hip
+    * A TargetCoreAttr describing the GPU features and limits in a single
+      GPU core (that is, AMD compute unit or NVIDIA streaming multiprocessor)
+    * An optional TargetChipAttr describing GPU features for the final chip
+      or product, e.g., core count
+  }];
+
+  let mnemonic = "target";
+  let cppNamespace = "::mlir::iree_compiler::IREE::GPU";
+
+  let parameters = (ins
+    "TargetAPI":$api,
+    StringRefParameter<"target architecture">:$arch,
+    "TargetCoreAttr":$core,
+    OptionalParameter<"TargetChipAttr">:$chip
+  );
+
+  let assemblyFormat = "`<` struct(params) `>`";
+
+  let extraClassDeclaration = [{
+    int getPreferredSubgroupSize() const {
+      return getCore().getSubgroupSizeChoices().asArrayRef().front();
+    }
+
+    bool supportSubgroupShuffle() const {
+      return bitEnumContainsAll(getCore().getSubgroup().getValue(),
+                                SubgroupOps::Shuffle);
+    }
+
+    std::optional<int> getCUDAComputeCapability() const;
+    // Returns true if this target supports TensoreCore MMA ops with TF32
+    // input types.
+    bool supportTF32InputMMAOps() const;
+    // Returns true if this target supports TensorCore synchronized MMA ops.
+    bool supportSyncMMAOps() const;
+  }];
+}
+
+def IREEGPU_AbbrTargetAttr : AttrDef<IREEGPU_Dialect, "AbbrTarget"> {
```

Comment:
`Abbr` is weird  - maybe `TargetAliasAttr` / `TargetNameAttr`?

Reply:
Done.